catalog/nato/stanag4559/nato-stanag4559-source/src/main/java/org/codice/alliance/nato/stanag4559/source/Stanag4559FilterDelegate.java

```diff
@@ -0,0 +1,528 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.nato.stanag4559.source;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+import org.codice.alliance.nato.stanag4559.common.GIAS.AttributeInformation;
+
+import ddf.catalog.filter.FilterDelegate;
+
+public class Stanag4559FilterDelegate extends FilterDelegate<String> {
+
+    private Stanag4559FilterFactory filterFactory;
+
+    public static final String EMPTY_STRING = "";
+
+    public static final String SQ = "'";
+
+    public static final String UTC = "UTC";
+
+    private static final String DATE_FORMAT = "yyyy/MM/dd HH:mm:ss";
+
+    private static String view;
+
+    private static Map<String, List<AttributeInformation>> queryableAttributes;
+
+    public Stanag4559FilterDelegate(Map<String, List<AttributeInformation>> queryableAttributes,
+            String view) {
+        this.view = view;
+        this.queryableAttributes = queryableAttributes;
+        filterFactory = new Stanag4559FilterFactory(queryableAttributes, view);
+    }
+
+    @Override
+    public String and(List<String> filters) {
+        String filter = filterFactory.buildAndFilter(filters);
+        return filter;
+    }
+
+    @Override
+    public String or(List<String> filters) {
+        String filter = filterFactory.buildOrFilter(filters);
+        return filter;
+    }
+
+    @Override
+    public String not(String filter) {
+        String newFilter = filterFactory.buildNotFilter(filter);
+        return newFilter;
+    }
+
+    @Override
+    public String propertyIsEqualTo(String propertyName, String literal, boolean isCaseSensitive) {
+        String filter = filterFactory.buildPropertyIsEqualTo(propertyName, SQ + literal + SQ);
+        return filter;
+    }
+
+    @Override
+    public String propertyIsEqualTo(String propertyName, Date literal) {
+        String filter = filterFactory.buildPropertyIsEqualTo(propertyName,
+                getStringFromDate(literal));
+        return filter;
+    }
+
+    @Override
+    public String propertyIsEqualTo(String propertyName, int literal) {
+        String filter = filterFactory.buildPropertyIsEqualTo(propertyName, EMPTY_STRING + literal);
```

Comment:
I'd argue for `String.valueOf(literal)`, since it more clearly represents the intent of the code.